grub.cfg-boot

```diff
@@ -0,0 +1,38 @@
+set default=0
+set timeout=3
+
+# load only kernel_status from the bootenv
+load_env --file /EFI/ubuntu/grubenv kernel_status
+
+set cmdline="console=ttyS0 console=tty1 panic=-1 systemd.gpt_auto=0 snapd_recovery_mode=run"
+
+if [ "$kernel_status" = "try" ]; then
+    # a new kernel got installed
+    set kernel_status="trying"
+    save_env kernel_status
+
+    # try to boot with try-kernel.img
+    menuentry "Run Ubuntu Core 20" {
+        search --label ubuntu-boot --set=ubuntu_boot
+        chainloader ($ubuntu_boot)/try-kernel.efi $cmdline
+    }
+elif [ "$kernel_status" = "trying" ]; then
+    # nothing cleared the "trying snap" so the boot failed
+    # we clear the mode and boot normally
+    set kernel_status=""
+    save_env kernel_status
+elif [ x"$kernel_status" != x"" ]; then
```

Comment:
Is the x technique needed in grub scripts? Or maybe we could simply test for non-emptiness using `-n`?

Reply:
Yes, I wasn't clear on that, I think we might be able to just use `-n`, let me check

Reply:
yes use `-n`